src/storage/src/hummock/sstable/sstable_object_id_manager.rs

```diff
@@ -191,6 +186,41 @@ impl SstableObjectIdManager {
     }
 }
 
+#[async_trait::async_trait]
+impl GetObjectId for Arc<SstableObjectIdManager> {
+    /// Returns a new SST id.
+    /// The id is guaranteed to be monotonic increasing.
+    async fn get_new_sst_object_id(&mut self) -> HummockResult<HummockSstableObjectId> {
+        self.map_next_sst_object_id(|available_sst_object_ids| {
+            available_sst_object_ids.get_next_sst_object_id()
+        })
+        .await
+    }
+}
+
+/// `SharedComapctorObjectIdManager` is used to get output sst id for serverless compaction.
+#[derive(Clone)]
+pub struct SharedComapctorObjectIdManager {
+    output_object_ids: VecDeque<u64>,
+}
+
+impl SharedComapctorObjectIdManager {
+    pub fn new(output_object_ids: VecDeque<u64>) -> Self {
+        Self { output_object_ids }
+    }
+}
+
+#[async_trait::async_trait]
+impl GetObjectId for SharedComapctorObjectIdManager {
+    async fn get_new_sst_object_id(&mut self) -> HummockResult<HummockSstableObjectId> {
+        if let Some(first_element) = self.output_object_ids.pop_front() {
+            Ok(first_element)
+        } else {
+            return Err(HummockError::meta_error("Output object id runs out"));
```

Comment:
It is werid to treat this as an meta error here since there is no meta or meta client involved.